flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -2064,11 +2101,19 @@ public void jobManagerLostLeadership(final JobID jobId, final JobMasterId jobMas
         public void handleError(Throwable throwable) {
             onFatalError(throwable);
         }
+
+        @Override
+        public void jobManagerRejectedRegistration(
+                JobID jobId, String targetAddress, JMTMRegistrationRejection rejection) {
+            runAsync(() -> handleRejectedJobManagerConnection(jobId, targetAddress, rejection));
```

Comment:
_should_ it actually clean up everything? Why is it not just asking for the actual leader, and only failing if that times out?